crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -340,8 +565,48 @@ impl<'db> OverloadBinding<'db> {
         }
     }
 
-    pub(crate) fn has_binding_errors(&self) -> bool {
-        !self.errors.is_empty()
+    fn as_result(&self) -> Result<(), CallErrorKind> {
+        if !self.errors.is_empty() {
+            return Err(CallErrorKind::BindingError);
+        }
+        Ok(())
+    }
+}
+
+/// Describes a callable for the purposes of diagnostics.
+#[derive(Debug)]
+pub(crate) struct CallableDescriptor<'a> {
```

Comment:
heh, "descriptor" is quite an overloaded word in red-knot 😆 maybe `CallableDescription`?

```suggestion
pub(crate) struct CallableDescription<'a> {
```

Reply:
Done